### ISSUE TITLE
specify markup format for 'doc' fields

### ISSUE THREAD

**Issue by tetron** [common-workflow-language/cwl-v1.3#36]
Doc fields should support markup.

Possible approaches:

* HTML only with a restricted tag set
* markdown (commonmark) (converts to HTML, can embed HTML)
* Something else?
* multiple markup languages (???), with a field that indicates what markup is in use

**Comment by mr-c** [comment]
cross-referencing via HTML anchors from the resolved schema-salad identifiers of this and all sub-documents?

see also https://github.com/common-workflow-language/common-workflow-language/issues/293

**Comment by tetron** [comment]
From discussion:  commonmark + HTML is the obvious choice. 

```
const domPurifyConfig: TDomPurifyConfig = {
    ALLOWED_TAGS: [
        'a',
        'b',
        'blockquote',
        'br',
        'code',
        'del',
        'dd',
        'dl',
        'dt',
        'em',
        'h1',
        'h2',
        'h3',
        'h4',
        'h5',
        'h6',
        'hr',
        'i',
        'img',
        'kbd',
        'li',
        'ol',
        'p',
        'pre',
        's',
        'del',
        'section',
        'span',
        'strong',
        'sub',
        'sup',
        'ul',
        'table',
        'thead',
        'tbody',
        'th',
        'tr',
        'td',
    ],
    ALLOWED_ATTR: ['src', 'width', 'height', 'href', 'alt', 'title', 'style' ],
};
```

**Comment by tetron** [comment]
> cross-referencing via HTML anchors from the resolved schema-salad identifiers of this and all sub-documents?
> 
> see also [common-workflow-language/common-workflow-language#293](https://github.com/common-workflow-language/common-workflow-language/issues/293)

Yes, should be possible to cross-reference between 'doc' sections using the document id generated by schema salad, with a recommendation that the implementation will have them be the identifiers or will be rewritten to be correct.
